### PR TITLE
allow forcing the non-use of the /ja endpoint

### PR DESCRIPTION
**Pull request by travisghansen**
- Option to disable /ja endpoint usage
- `lock` for thread safety

### CODE REVIEW

File: pyopensprinkler/__init__.py [hunk 1]
```diff
@@ -67,6 +68,18 @@ def __init__(self, url, password, opts=None):
         self._programs = {}
         self._stations = {}
         self._state = None
+        self._use_ja_endpoint = os.environ.get("PYOPENSPRINKLER_USE_JA_ENDPOINT", None)
```
Comment (vinteo) [inline pyopensprinkler/__init__.py:71]: I think the option should be flipped the other way to `PYOPENSPRINKLER_NOT_USE_JA_ENDPOINT` or `PYOPENSPRINKLER_SKIP_JA_ENDPOINT` cause the current name makes it seems like the default is to not use the /ja endpoint which is not the case. 

Maybe also better to be `PYOPENSPRINKLER_SKIP_GROUPED_ENDPOINT`  or `PYOPENSPRINKLER_SKIP_ALL_ENDPOINT`

File: pyopensprinkler/__init__.py [hunk 2]
```diff
@@ -161,26 +174,37 @@ def refresh(self):
...
-                content = {
-                    "settings": settings,
-                    "options": options,
-                    "stations": stations,
-                    "status": status,
-                    "programs": programs,
-                }
-            else:
-                raise exc
+        use_ja = True
```
Comment (vinteo) [inline pyopensprinkler/__init__.py:177]: I think here we can check for state if not none to check for firmware version to default to False if the firmware < 216 so it does try /ja except for the first time

File: pyopensprinkler/__init__.py [hunk 3]
```diff
@@ -161,26 +174,37 @@ def refresh(self):
...
+        if use_ja:
+            try:
+                (_, content) = self.request("/ja")
+                self._state = content
+                return
+            except OpenSprinklerApiError as exc:
+                (_, err_code) = exc.args
+                if err_code == 32:
+                    # set for preemptive behavior on all subsequent calls
+                    self._use_ja_endpoint = False
```
Comment (vinteo) [inline pyopensprinkler/__init__.py:190]: I guess this is similar to the firmware check above although I think the firmware check is more explicit.

File: pyopensprinkler/__init__.py (hunk 2 above)
Comment (travisghansen) [reply pyopensprinkler/__init__.py:177]: It does this behavior already actually.

File: pyopensprinkler/__init__.py (hunk 3 above)
Comment (travisghansen) [reply pyopensprinkler/__init__.py:190]: Yeah, the firmware check is a chicken/egg IMO...is there ever any scenario where a `32` would come back on firmware that supports the /ja endpoint?

Maybe we force people to set the firmware version when instantiating the controller?

File: pyopensprinkler/__init__.py (hunk 1 above)
Comment (travisghansen) [reply pyopensprinkler/__init__.py]: Went with the last `PYOPENSPRINKLER_SKIP_ALL_ENDPOINT`.

**Pull request merged**